cranker/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+        rpc_filter::{ Memcmp, RpcFilterType },
+        nonce_utils::get_account,
+    },
+    std::{ sync::Arc, time::{ Duration, SystemTime, UNIX_EPOCH } },
+    tokio::time,
+    tracing::{ info, error },
+    stake_deposit_interceptor::{
+        state::{
+            DepositReceipt,
+            StakePoolDepositStakeAuthority,
+            StakeDepositInterceptorDiscriminators,
+        },
+        instruction::create_claim_pool_tokens_instruction,
+    },
+    spl_associated_token_account::get_associated_token_address,
+    std::thread,
+    solana_program::program_pack::Pack,
+    spl_token::state::Account as TokenAccount,
+};
+
+use spl_associated_token_account::instruction::create_associated_token_account;
+
+#[derive(Clone)]
+pub struct CrankerConfig {
+    pub rpc_url: String,
+    pub ws_url: String,
+    pub program_id: Pubkey,
+    pub payer: Arc<Keypair>, // Wrapped in Arc
+    pub interval: Duration,
+    pub commitment: CommitmentConfig,
+}
+
+pub struct InterceptorCranker {
+    rpc_client: Arc<RpcClient>,
+    payer: Arc<Keypair>,
+    program_id: Pubkey,
+    metrics: Arc<std::sync::Mutex<Metrics>>,
+    interval: Duration,
+}
+
+impl InterceptorCranker {
+    pub fn new(config: CrankerConfig) -> Self {
+        let rpc_client = Arc::new(
+            RpcClient::new_with_commitment(config.rpc_url, config.commitment)
+        );
+
+        Self {
+            rpc_client,
+            payer: config.payer, // No need to clone Arc
+            program_id: config.program_id,
+            metrics: Arc::new(std::sync::Mutex::new(Metrics::default())),
+            interval: config.interval, // Store the interval
+        }
+    }
+
+    pub async fn start(&self) {
+        info!("Starting InterceptorCranker service");
+        let mut interval_timer = time::interval(self.interval);
+        info!("Set interval timer to {} seconds", self.interval.as_secs());
+
+        loop {
+            interval_timer.tick().await;
+            info!("Tick: Starting new processing cycle");
+
+            match self.process_expired_receipts().await {
+                Ok(_) => info!("Successfully processed expired receipts"),
+                Err(e) => error!("Error processing receipts: {}", e),
+            }
+        }
+    }
+
+    async fn process_expired_receipts(&self) -> Result<(), CrankerError> {
+        info!("Starting to process expired receipts");
+        let receipts = self.get_deposit_receipts().await?;
+        info!("Found {} deposit receipts", receipts.len());
+
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map_err(|e| CrankerError::TimeError(e.to_string()))?
+            .as_secs();
+
+        for receipt in receipts {
+            // Get raw bytes using bytemuck and interpret as little-endian
+            let deposit_time_bytes = bytemuck::bytes_of(&receipt.deposit_time);
+            let deposit_time = u64::from_le_bytes(deposit_time_bytes.try_into().unwrap());
```

Comment:
with the PodU64, I believe you could call `u64::from()` directly on the property like `u64::from(receipt.deposit_time)`